src/mango/src/mango_idx_view.erl

```diff
@@ -54,7 +54,8 @@ add(#doc{body={Props0}}=DDoc, Idx) ->
     NewView = make_view(Idx),
     Views2 = lists:keystore(element(1, NewView), 1, Views1, NewView),
     Props1 = lists:keystore(<<"views">>, 1, Props0, {<<"views">>, {Views2}}),
-    {ok, DDoc#doc{body={Props1}}}.
+    Props2 = Props1 ++ [{<<"autoupdate">>, false}],
```

Comment:
This is so that we don't include query views via ken because of in-transaction updates, right? Is this a bug waiting to happen if someone creates a query design doc by hand or via replication?

Reply:
Correct on which bit? Are there bugs here that you are trying to prevent by disabling new ken? If so this certainly isn't the correct approach as there are plenty of ways to get design docs into a db outside this function.

New ken is index agnostic which means it's the wrong place to add such a thing. You could maybe patch couch_views to not auto-build mango indexes but I'd very much want to reconsider whether that's a bug in fighting between the doc transactions and view update transactions. Though we'd likely want to still find a way for background indexers to skip updates maybe?

Bottom line, I'm fine doing a thing to avoid wasting work on updates that have already happened, but its way more important to narrow down whether simultaneous updates would break an index or not.